aiperf/records/records_manager.py

```diff
@@ -113,18 +117,43 @@ def __init__(
         self.worker_stats_lock: asyncio.Lock = asyncio.Lock()
 
         self._previous_realtime_records: int | None = None
+        self._previous_realtime_telemetry_samples: int = 0
 
-        # Telemetry data storage
-        self._telemetry_hierarchy = TelemetryHierarchy()
-        self._telemetry_hierarchy_lock = asyncio.Lock()
+        # Telemetry error tracking
         self._telemetry_error_counts: dict[
             ErrorDetails, int
         ] = {}  # Track telemetry-specific errors with counts
         self._telemetry_error_counts_lock = asyncio.Lock()
+        self._telemetry_warnings_logged: set[str] = set()
+        self._telemetry_endpoints_contacted: set[str] = set()
+
+        # Performance tracking for telemetry
+        self._telemetry_task_runs: int = 0
+        self._telemetry_total_gen_time_ms: float = 0.0
+        self._telemetry_total_pub_time_ms: float = 0.0
+        self._telemetry_total_metrics_generated: int = 0
+        self._telemetry_mode_enabled_time: float | None = None
```

Comment:
im thinking maybe a lot of this should be moved to a locally defined @dataclass to keep it tidy?